src/Storefront/Page/Account/Login/AccountLoginPageLoader.php

```diff
@@ -95,7 +95,7 @@ private function getCountries(SalesChannelContext $salesChannelContext): Country
     {
         $criteria = (new Criteria())
             ->addFilter(new EqualsFilter('active', true))
-            ->addAssociation('country.states');
+            ->addAssociation('states');
```

Comment:
the `country.` prefix should be removed in `\Shopware\Core\Framework\DataAbstractionLayer\Dbal\EntityReader::addAssociationFieldsToCriteria()`, so the change should not be necessary.

Reply:
Something is not clean, I cannot tell what exactly though.

My first instinct is: It is weird that Criteria referencing only one Definition uses the "associative filtering" at all. I don't know if this should work, which is mainly a design-decision I guess. I would recommend against allowing that, because I do not see any gain from that.

If this should be allowed nevertheless, then it is still weird. The call stack going down to the exception/log message I introduced in #2204 comes from here:
https://github.com/shopware/platform/blob/aca1e5c4691dd72b0a8cfb1ae7d46e0800f16222/src/Core/Framework/DataAbstractionLayer/Dbal/EntityReader.php#L66-L80
The line `$definition->getFields()->getBasicFields()` contains `[ "country" ]`. Why is that, since its neither a field, nor an existing association and actually an association which should be cleaned up somewhere along the way?

Reply:
> I don't know if this should work, which is mainly a design-decision I guess
Yes it is intended to work like this, IMHO there is no real gain too, but removing that "feature" would be a breaking change. And besides that the break would be subtle as no static analysis would pick that up, so changing that would mean that everybody needs to check the way they add Associations to every criteria (not just on php side but also on admin side or when using the API)

> The line $definition->getFields()->getBasicFields() contains [ "country" ]
That's weird as i cannot reproduce that. `$definition` is a `CountryDefinition` and there are no plugins active that may add fields? What kind of field is the `country` field?

Reply:
> Yes it is intended to work like this

Well, but it doesn't?
<img width="567" alt="Bildschirmfoto 2021-12-18 um 12 33 51" src="https://user-images.githubusercontent.com/18024904/146639524-849599a3-c20e-4b3d-a4f5-2f8abf6704a2.png">
This is content of `country_state`:
<img width="656" alt="Bildschirmfoto 2021-12-18 um 12 34 09" src="https://user-images.githubusercontent.com/18024904/146639538-dc9c88a1-9bad-489c-9887-d670ecf47296.png">

Whereas, when I remove `country` from the association as proposed in this PR, then I get:
<img width="551" alt="Bildschirmfoto 2021-12-18 um 12 40 41" src="https://user-images.githubusercontent.com/18024904/146639647-397fde6e-d555-428e-b88e-c579f3bd6b05.png">

Reply:
Ah ok, now i get it and actually you are right here. Adding associations prefixed with the root entity name is not supported. That is only supported in filters.